kotlinx-coroutines-debug/src/internal/DebugProbesImpl.kt

```diff
@@ -219,23 +260,47 @@ internal object DebugProbesImpl {
     internal fun probeCoroutineSuspended(frame: Continuation<*>) = updateState(frame, State.SUSPENDED)
 
     private fun updateState(frame: Continuation<*>, state: State) {
-        if (!isInstalled) return
+        // KT-29997 is here only since 1.3.30
+        if (state == State.RUNNING && KotlinVersion.CURRENT.isAtLeast(1, 3, 30)) {
+            updateRunningState(frame, state)
+            return
+        }
+
         // Find ArtificialStackFrame of the coroutine
-        val owner = frame.owner()
+        val owner = frame.owner() ?: return
         updateState(owner, frame, state)
     }
 
+    @Synchronized // See comment to infoCache
+    private fun updateRunningState(continuation: Continuation<*>, state: State) {
```

Comment:
It would be more consistent to always name the parameter `frame` in all `updateState` functions. Also, consider using `frame: CoroutineStackFrame` type as a parameter in all internal function as they all seem to contain the same line of code: `val frame = continuation as? CoroutineStackFrame ?: return`

Reply:
I've done it only for this method.
Otherwise code becomes less readable: somewhere `Continuation<*>` is required, `updateState` is ok with nullable `CoroutineStackFrame` etc.